org.eclipse.jdt.core/model/org/eclipse/jdt/core/compiler/CompilationParticipant.java

```diff
@@ -143,6 +146,28 @@ public void processAnnotations(BuildContext[] files) {
 	// do nothing by default
 }
 
+/**
+ * Notifies this participant about a finished class compilation. This
+ * notification occurs right before the compiled class is written out to disk.
+ * This allows to perform arbitrary byte code manipulation on the generated
+ * class.
+ *
+ * There is no specific order in which potential multiple compilation participants are called. The received class data
```

Comment:
Please start with "If the "requires" attribute is not used for participant definition, there will be no specific order ..."